sr_ronex_controllers/src/spi_base_controller.cpp

```diff
@@ -95,50 +106,55 @@ void SPIBaseController::update(const ros::Time&, const ros::Duration&)
 {
   for (uint16_t spi_index = 0; spi_index < NUM_SPI_OUTPUTS; ++spi_index)
   {
+    // Check if the status has been processed
+    if (delete_status_[spi_index] && status_queue_[spi_index].size() > 0)
+    {
+      status_queue_[spi_index].pop();
+      delete_status_[spi_index] = false;
+    }
     // Check if we need to update a status
     if ( status_queue_[spi_index].size() > 0)
     {
-      if ( status_queue_[spi_index].front().second == NULL )
+      // check if two cycles has passed since the cmd was received so it contains its answer
+      if (loop_count_ == status_queue_[spi_index].front().second.loop_number + 2 )
       {
-        if (new_command)
-        {
-          new_command = false;
-          spi_->nullify_command(spi_index);
-          continue;
-        }
-
         // the response has not been received. If the command type is NORMAL
         // then the response can be updated (it's INVALID until the SPI responds)
         if ( spi_->state_->command_type == RONEX_COMMAND_02000002_COMMAND_TYPE_NORMAL )
         {
-          status_queue_[spi_index].front().second =
-                  new SPI_PACKET_IN(spi_->state_->info_type.status_data.spi_in[spi_index]);
+          status_queue_[spi_index].back().second.received = true;
+          status_queue_[spi_index].back().second.packet =
+              SPI_PACKET_IN(spi_->state_->info_type.status_data.spi_in[spi_index]);
```

Comment:
I think we don't need to use the constructor here, as the assignment will do a copy of the whole structure anyway. We don't need an intermediate copy.

```cpp
status_queue_[spi_index].back().second.packet = spi_->state_->info_type.status_data.spi_in[spi_index];
```